users/tests/test_logout_view.py

```diff
@@ -90,3 +90,138 @@ def test_logout_redirect_next_authenticated(client, user_factory, application_fa
+    )
+    settings.SOCIAL_AUTH_DUMMYOIDCBACKEND_REDIRECT_LOGOUT_TO_END_SESSION = True
+
+    application_factory(
+        post_logout_redirect_uris='http://example.com/',
+        redirect_uris=['http://example.com/']
+    )
+
+    password = get_random_string()
+    user = user_factory(password=password)
+
+    usersocialauth_factory(provider='dummyoidcbackend', user=user)
+
+    client.login(username=user.username, password=password)
+
+    params = {'post_logout_redirect_uri': 'http://example.com/'}
+    response = client.get('/openid/end-session', params, follow=False)
+
+    assert response.status_code == 302
+    assert response.url == (
+        'http://dummyoidcbackend.example.com/openid/end-session?'
+        'post_logout_redirect_uri=http%3A%2F%2Ftestserver%2Fopenid%2Fend-session'
+    )
+
+    response = client.get('/openid/end-session', follow=False)
+
+    assert response.status_code == 200
+    assert link_to_url_found_in_response(response, 'http://example.com/')
+
+
+@pytest.mark.django_db
+def test_logout_redirect_to_two_third_party_oidc_end_sessions(
+    settings,
+    client,
+    application_factory,
+    user_factory,
+    usersocialauth_factory,
+    assertCountEqual,
+):
+    settings.AUTHENTICATION_BACKENDS = settings.AUTHENTICATION_BACKENDS + (
+        'users.tests.conftest.DummyOidcBackend',
+        'users.tests.conftest.DummyOidcBackend2',
+    )
+    settings.SOCIAL_AUTH_DUMMYOIDCBACKEND_REDIRECT_LOGOUT_TO_END_SESSION = True
+    settings.SOCIAL_AUTH_DUMMYOIDCBACKEND2_REDIRECT_LOGOUT_TO_END_SESSION = True
+
+    application_factory(
+        post_logout_redirect_uris='http://example.com/',
+        redirect_uris=['http://example.com/']
+    )
+
+    password = get_random_string()
+    user = user_factory(password=password)
+
+    usersocialauth_factory(provider='dummyoidcbackend', user=user)
+    usersocialauth_factory(provider='dummyoidcbackend2', user=user)
+
+    client.login(username=user.username, password=password)
+
+    response_urls = []
+    expected_response_urls = [
+        (
+            'http://dummyoidcbackend.example.com/openid/end-session?'
+            'post_logout_redirect_uri=http%3A%2F%2Ftestserver%2Fopenid%2Fend-session'
+        ),
+        (
+            'http://dummyoidcbackend2.example.com/openid/end-session?'
+            'post_logout_redirect_uri=http%3A%2F%2Ftestserver%2Fopenid%2Fend-session'
+        )
+    ]
+
+    params = {'post_logout_redirect_uri': 'http://example.com/'}
+    response = client.get('/openid/end-session', params, follow=False)
+
+    assert response.status_code == 302
+    assert response.url in expected_response_urls
+    response_urls.append(response.url)
+
+    response = client.get('/openid/end-session', params, follow=False)
+
+    assert response.status_code == 302
+    assert response.url in expected_response_urls
+    response_urls.append(response.url)
+
+    assertCountEqual(response_urls, expected_response_urls)
```

Comment:
This is not a biggy, but if I understand this correctly, the redirects can happen in any order and that's why the responses' urls can't be asserted in the simplest way. If my understanding is right, then how about using sets? So, make `response_urls` and `expected_response_urls` sets, collect response urls into `response_urls` (just like currently), and finally `assert response_urls == expected_response_urls`.

Reply:
True. Was a quick fix and didn't think about it.